src/notify.js

```diff
@@ -71,6 +83,11 @@ Notify.isSupported = function(perm) {
         throw new Error('You must only call this before calling Notification.requestPermission(), otherwise this feature detect would trigger an actual notification!');
     }
 
+    if (Notify.serviceWorkerSupported()) {
```

Comment:
I think we should split this method out into two separate feature detects, one for `isWebNotificationSupported()` and another for `isPushNotificationSupported()`? I think the two are different enough that developers may want to distinguish when one is supported and not another?